strutil/pathiter.go

```diff
@@ -82,6 +82,11 @@ func (iter *PathIterator) CurrentPath() string {
 	return iter.path[:iter.right]
 }
 
+// CurrentCleanPath returns the same value as CurrentPath with the right slash trimmed.
+func (iter *PathIterator) CurrentCleanPath() string {
```

Comment:
This is also used in `Part 4` for the new variation of MkdirAllWithin.

Reply:
maybe calling it CurrentPathNoSlash would be clearer?

Reply:
I changed all pathiter method that "clean" to ...NoSlash